mlrun/artifacts/base.py

```diff
@@ -1276,18 +1281,35 @@ def get_artifact_meta(artifact):
 
 
 def generate_target_path(item: Artifact, artifact_path, producer):
-    # path convention: artifact_path[/{run_name}]/{iter}/{key}.{suffix}
+    # path convention: artifact_path[/{run_name}]/{iter}/src_path/{key}.{suffix}
     # todo: add run_id here (vs in the .run() methods), support items dedup (by hash)
     artifact_path = artifact_path or ""
-    if artifact_path and not artifact_path.endswith("/"):
-        artifact_path += "/"
+    if artifact_path and not artifact_path.endswith(os.sep):
+        artifact_path += os.sep
```

Comment:
I'm not sure using `os.sep` is the right thing to do, as it will produce the separator of the OS you're using where this runs. So, if this code runs on your Windows laptop (as this is client code), the artifact path will be created with `\\` as separators and this will most likely not work for most targets (well, everything beside Windows, which I don't think there's any way in which we use as target). So, I would keep the `/` here.

Reply:
gotcha, its an old habit of mine, I'll adapt :)